packages/core/src/Card.js

```diff
@@ -6,20 +6,35 @@ import { applyVariations, getPaletteColor, deprecatedColorValue } from './utils'
 
 const boxShadow = props => {
   const boxShadows = {
-    sm: {
-      'box-shadow': props.theme.boxShadows[0]
-    },
-    md: {
-      'box-shadow': props.theme.boxShadows[1]
-    },
-    lg: {
-      'box-shadow': props.theme.boxShadows[2]
-    },
-    xl: {
-      'box-shadow': props.theme.boxShadows[3]
+    sm: props.theme.boxShadows[0],
+    md: props.theme.boxShadows[1],
+    lg: props.theme.boxShadows[2],
+    xl: props.theme.boxShadows[3]
+  }
+
+  if (typeof props.boxShadowSize == 'string') {
+    return {
+      'box-shadow': boxShadows[props.boxShadowSize]
     }
   }
-  return boxShadows[props.boxShadowSize]
+
+  return `
+    @media only screen and (min-width: 520px) {
+      box-shadow: ${boxShadows[props.boxShadowSize[0] || 'sm']}
+    }
+​
+    @media only screen and (min-width: 768px) {
+      box-shadow: ${boxShadows[props.boxShadowSize[1] || 'md']}
+    }
+​
+    @media only screen and (min-width: 992px) {
+      box-shadow: ${boxShadows[props.boxShadowSize[2] || 'lg']}
+    }
+​
+    @media only screen and (min-width: 1200px)  {
+      box-shadow: ${boxShadows[props.boxShadowSize[3] || 'xl']}
```

Comment:
If I provide the prop `boxShadowSize={['md', 'sm']}` I would expect medium boxShadows on the smallest breakpoint and the small boxShadows on all other breakpoints, I don't think this behaviour would occur with the current implementation. Unfortunately I don't immediately have an idea for how to write this in a clean way.

Additionally, I think we would want to use the theme breakpoints rather than hardcoded breakpoints.

Reply:
I agree with @James300 - hardcoded breakpoints are a non-starter

Reply:
or @pcln-james as he is now known

Reply:
Just realized this relates to #844; whatever approach we take will probably apply to both `Card` and `Box`, but we might want to see if Styled System has any utility for this. I know it has a `boxShadow` prop already that supports the responsive array, but I'm not sure how that fits with the `boxShadows` values on our theme.